offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/projects/ProjectMainConnector.groovy

```diff
@@ -0,0 +1,134 @@
+package life.qbic.portal.offermanager.dataresources.projects
+
+import groovy.util.logging.Log4j2
+import life.qbic.datamodel.dtos.general.Person
+import life.qbic.business.projects.spaces.CreateProjectSpaceDataSource
+import life.qbic.business.projects.spaces.ProjectSpaceExistsException
+import life.qbic.business.projects.create.CreateProjectDataSource
+import life.qbic.business.projects.create.ProjectExistsException
+
+import life.qbic.datamodel.dtos.business.*
+import life.qbic.datamodel.dtos.projectmanagement.*
+
+import life.qbic.business.exceptions.DatabaseQueryException
+
+import java.sql.Connection
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.sql.Statement
+
+import life.qbic.openbis.openbisclient.OpenBisClient
+
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.operation.IOperation;
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.project.create.CreateProjectsOperation;
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.project.create.ProjectCreation;
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.project.id.ProjectIdentifier;
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.space.create.CreateSpacesOperation;
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.space.create.SpaceCreation;
+import ch.ethz.sis.openbis.generic.asapi.v3.dto.space.id.SpacePermId;
+import ch.ethz.sis.openbis.generic.asapi.v3.IApplicationServerApi;
+
+/**
+ * Provides operations on QBiC project data
+ *
+ * This class implements the data sources of the different use cases and is responsible for transferring data to the customer db and openBIS
+ *
+ * @since: 1.0.0
+ *
+ */
+@Log4j2
+class ProjectMainConnector implements CreateProjectDataSource, CreateProjectSpaceDataSource {
+
+  /**
+   * A connection to the customer database used to create queries.
+   */
```

Comment:
what customer database?

Reply:
changed wording, should make more sense now